runtime/src/accounts_db.rs

```diff
@@ -2254,8 +2254,10 @@ impl AccountsDb {
             if let Some(slot_stores) = self.storage.get_slot_stores(slot) {
                 slot_stores.write().unwrap().retain(|_key, store| {
                     if store.count() == 0 {
-                        self.dirty_stores
-                            .insert((slot, store.append_vec_id()), store.clone());
+                        if !is_startup {
```

Comment:
admittedly, this is another hack. xD

glad that we have managed to revive the almost-dying flag (`_`-prefixed) for quick win. :)